### PR TITLE
RDoc-1873

### PR DESCRIPTION
**Pull request by reebhub**
document solution to cluster certificate renewal failure

### CODE REVIEW

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# Let's Encrypt: Solve Certificate Renewal Issue
+
+* If creating or renewing a cluster certificate fails and the
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:3]: This is not the symptom. Creating or renewing doesn't fail.
The symptom is when external replication or ETL to another cluster fails with permission errors.

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 2]
```diff
@@ -0,0 +1,151 @@
+# Let's Encrypt: Solve Certificate Renewal Issue
+
+* If creating or renewing a cluster certificate fails and the 
+  problem relates to [Public Key Pinning Hash](../../../server/security/authentication/certificate-renewal-and-rotation#implicit-trust-by-public-key-pinning-hash),
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:4]: The user most likely doesn't know what a Public Key Pinning Hash is, and they won't know it's related to that. 
The user will get permission errors for their External Replication and ETL tasks to another RavenDB cluster.

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 3]
```diff
@@ -0,0 +1,151 @@
+# Let's Encrypt: Solve Certificate Renewal Issue
+
+* If creating or renewing a cluster certificate fails and the 
+  problem relates to [Public Key Pinning Hash](../../../server/security/authentication/certificate-renewal-and-rotation#implicit-trust-by-public-key-pinning-hash), 
+  this article may help.  
+
+* When you create or renew cluster certificates, and `Let's Encrypt`'s certificate
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:7]: I would write:
If you have External Replication or ETL to another cluster or if you use your own Let's Encrypt certificates as client certificates, the next certificate renewal may cause permission issues that need to be handled manually.

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  Once the source cluster renewes its certificate, the destination cluster would 
+  no longer trust it because the thumbprint has changed.  
+  To solve this, the admin had to access the destination cluster and manually specify 
+  that it can trust the new source certificate.  
+
+* **Original Solution**  
+  To solve this problem, we introduced [Implicit Trust](../../../server/security/authentication/certificate-renewal-and-rotation#implicit-trust-by-public-key-pinning-hash) 
+  by using the certificate's **Public Key Pinning Hash**.  
+  Now, if the source cluster renews its certificate by using the same private key and 
+  issuer, the new certificate will have the same Public Key Pinning Hash, and the
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:57]: I think we need to highlight the fact that it needs the same private key and 
  issuer

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+As a result, RavenDB users that use the Public Key Pinning Hash solution need to 
+take a few manual steps.  
+
+This is relevant for you -  
+
+1. If you use RavenDB ETL or perform External Replication between two RavenDB clusters.  
+     * If you use RavenDB Cloud where certificates are automatically renewed.  
+     * If you used our Setup Wizard with Let's Encrypt, and certificates are auto-renewed.  
+     * If you used your own Let's Encrypt certificate, and you renew it yourself with
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:82]: your own Let's Encrypt cluster certificate, and...

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+The hash in the example is the Public Key Pinning Hash of the Let's Encrypt X3 intermediate 
+certificate authority which was previously used to sign Let's Encrypt certificates.  
+
+This solution is stronger, it ensures a smooth transparent transition when the source cluster 
+certificate will renew itself and the issuer will actually change. The advantage here is that 
+there is no downtime.  
+
+The disadvantage is that an admin needs to access the machines themselves. They need to edit 
+the settings.json file on all nodes and restart the RavenDB service (one-by-one for no downtime).
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:108]: node-by-node

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+###Solution 2:
+
+1. In the case of ETL or External Replication, go to the source cluster and renew the Let's Encrypt 
+   cluster certificate. Then export it and register it in the destination cluster.  
+
+     * Click `Renew` on the server certificate in one of the nodes of the source cluster.  
+       ![Figure 1. Renew Server Certificate](images/renew_server_certificate.png)
+     * Export the new cluster certificate:  
+       ![Figure 2. Export Cluster Certificate](images/export_cluster_certificates.png)
+     * Copy the exported PFX file to one of the nodes in the destination cluster and upload
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:121]: No need to copy the file. Change to:
In the destination cluster studio, upload the exported pfx file as a client certificate,

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+     * Copy the exported PFX file to one of the nodes in the destination cluster and upload 
+       it as a client certificate:  
+       ![Figure 3. Upload Client Certificat](images/upload-client-certificate.png)
+
+
+2. In case you are using your own Let's Encrypt client certificates, simply renew them 
+   (on your own) and re-register them in the certificate view in the Studio ([Upload client 
+   certificate](../../../studio/database/tasks/import-data/import-from-ravendb#step-#1:-prepare-servers-for-the-import-process-(secure-4.x-servers-only))).  
+
+     This solution is easier and doesn't require access to the machines themselves, just an admin
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:130]: The following lines are for the entire solution 2, for both sections (1 and 2) so the indentation is wrong

File: Documentation/5.1/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown [hunk 9]
```diff
@@ -0,0 +1,368 @@
...
+* "ExternalIp" should be defined only if you are running behind a firewall (cloud VM, docker, etc...).  
+
+If all of this looks right, and the certificate still doesn't renew automatically and there are no alerts telling you what's wrong, 
+you can contact support. Make sure to supply the server logs with your ticket. When running in a cluster, please provide the logs from all nodes.
+
+If your logs are turned off, open `Manage Server`->`Admin Logs` in the Studio, and keep them open while you click the `Renew` button in the certificate view.
+
+### Creating or Renewing Let's Encrypt certificate fails
+
+If creating or renewing a cluster certificate fails and the problem relates
```
Comment (iftahbe) [inline Documentation/5.1/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown:152]: Same comment as above, these are not the symptoms.

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:3]: fixed

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 2 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:4]: fixed

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:7]: fixed

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 4 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:57]: done

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 5 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:82]: fixed

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 6 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:108]: fixed

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 7 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:121]: done

File: Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown (hunk 8 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/authentication/solve-cluster-certificate-renewal-issue.markdown:130]: done

File: Documentation/5.1/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown (hunk 9 above)
Comment (reebhub) [reply Documentation/5.1/Raven.Documentation.Pages/server/security/common-errors-and-faq.markdown:152]: fixed

**Review by iftahbe — APPROVED** [review]

**Pull request merged**